### ISSUE TITLE
Tests fail on Python 3.8 and 3.9 due to upstream issue in `distrax`

### ISSUE THREAD

**Issue by matt-graham** [kazewong/flowMC#100]
Raising issue as part of JOSS review https://github.com/openjournals/joss-reviews/issues/5021

Running the tests in clean Python 3.8 or 3.9 environments I get a series of
```
TypeError: Subscripted generics cannot be used with class and instance checks
```
errors which appear to be due to deepmind/distrax/issues/224. It looks like the most recent 0.1.3 release of `distrax` on PyPI fixes these issues as manually updating to that version and rerunning the tests in the Python 3.8 and Python 3.9 environments fixes the errors, so just bumping the pinned version of `distrax` to 0.1.3 in `setup.cfg` should fix.

**Comment by matt-graham** [comment]
It may also be worth changing the Python versions tested over in the GitHub Actions workflow 
https://github.com/kazewong/flowMC/blob/239c737856c5f3c3bce473f973253580a8cb85c3/.github/workflows/python-package.yml#L19
to include Python 3.8 and 3.9 to automatically catch these sort of problems

**Comment by kazewong** [comment]
@matt-graham Thanks for the catch. We saw the same problem at some point and I thought I had fixed it by bumping `distrax` to version 0.1.3, but the commit must have been lost somewhere. 

Anyway, this is now fixed in #102. The automated tests now runs on python 3.8, 3.9 3.10 and they all passes.
